### PR TITLE
feat(frontend): UI overhaul

### LINKED ISSUE #3447: [Bug]: LoadPreviousSessionModal is always appearing in App.tsx
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://docs.all-hands.dev/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

Upon opening OD in the browser the modal to restart/start a session is always appearing, regardless of _actually_ a session being available or not.
The current only check is depending on a token being present, but that is _always_ being passed in.

Ideally, if there is no actual session, a new session should be started automatically, especially if storage is set to memory.

### Current OpenDevin version

```bash
0.8.3
```

### Installation and Configuration

```bash
any
```

### Model and Agent

any

### Operating System

any

### Reproduction Steps

Just open OD in browser

### Logs, Errors, Screenshots, and Additional Context

_No response_

### PR DESCRIPTION
**Pull request by amanape**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**
Extends and improves the UI based on Paul's new and approved designs.

---

**CHANGELOG**
- Setup React Router
- Created a new Home screen
- Main app is in `/app`
- Rework settings modal

---

Resolves #3447, #3755, #3584, #3683, #3736

### CODE REVIEW

**Comment by tobitege** [comment]
This is exciting!! 😍

**Comment by mamoodi** [comment]
Just want to make sure you are aware of this PR which changes some things in the Settings:
https://github.com/All-Hands-AI/OpenHands/pull/3751

**Comment by rbren** [comment]
@amanape might be helpful to attach a few screenshots so folks can see what we're up to!

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (rbren) [inline package-lock.json:1]: This should be removed

File: openhands/server/listen.py [hunk 2]
```diff
@@ -55,14 +57,18 @@
 from openhands.server.auth import get_sid_from_token, sign_token
 from openhands.server.session import SessionManager
 
+from dotenv import load_dotenv
+
+load_dotenv()
```
Comment (rbren) [inline openhands/server/listen.py:62]: why is this here?

File: openhands/server/listen.py [hunk 3]
```diff
@@ -754,4 +788,61 @@ async def security_api(request: Request):
...
+        zip_file = zip_directory(workspace_dir)
+        response = FileResponse(zip_file, filename="workspace.zip", media_type='application/x-zip-compressed')
+
+        return response
+    except Exception as e:
+        logger.error(f'Error zipping workspace: {e}', exc_info=True)
+        raise HTTPException(
+            status_code=500,
+            detail='Failed to zip workspace',
+        )
```
Comment (rbren) [inline openhands/server/listen.py:807]: The zip needs to take place inside the runtime, e.g. by sending a `run_command` and then copying the resulting file. That's the only way this will work with a remote runtime

File: openhands/server/.env.sample [hunk 4]
```diff
@@ -0,0 +1,2 @@
+GITHUB_CLIENT_ID=""
```
Comment (rbren) [inline openhands/server/.env.sample:1]: Let's remove this

File: openhands/runtime/client/runtime.py [hunk 5]
```diff
@@ -289,7 +289,7 @@ def _init_container(
                 name=self.container_name,
                 detach=True,
                 environment=environment,
-                volumes=volumes,
+                volumes=None,
```
Comment (rbren) [inline openhands/runtime/client/runtime.py:292]: We need to keep this, but should only set `volumes` if the right env vars are set (e.g. `workspace_mount_path`)

File: openhands/server/listen.py (hunk 2 above)
Comment (amanape) [reply openhands/server/listen.py]: For the SaaS mode GitHub OAuth env variables. The last endpoint in `openhands/server/listen.py` (`'/github/callback'`) needs them to retrieve the `access_token`

**Comment by tobitege** [comment]
Huge kudos on this massive work, thank you!

Some small nits thus far:
- opened OH with this build for first time showed my LLM config in Advanced mode,
  but openening the Settings with the left-hand cog-wheel opened in "standard" mode
- the resizing grips will be re-added later? i.e. between chat column and workspace, between terminal and tabs
- all icons in the left sidebar should have tooltips
- more of a "long time" quirk: the "send message" icon looks like an "upload" image to me, is there something closer to resembling an ENTER key?
- the "file-input" opens a dialog with a graphics filter that includes like a dozen formats (Firefox); which I'm not sure if we could/should limit that to the most common image formats?
- the assistant's chat replies do not get any border (anymore)?
- the top-left OpenHands icon click asks for "Exit" confirmation, which seems specific to SaaS installation?

Independent of this PR:
I forgot to start Docker Desktop before sending a chat message and the backend runs into unrecoverable error (have to reload page at least and of course start docker):
```sh
10:56:01 - openhands:ERROR: runtime.py:203 - Launch docker client failed. Please make sure you have installed docker and started docker desktop/daemon.
10:56:01 - openhands:ERROR: agent_session.py:196 - Runtime initialization failed: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
10:56:01 - openhands:ERROR: agent_session.py:84 - Error starting session: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
```

**Comment by tobitege** [comment]
- Start screen says "Attach a file" -> "Attach an image"?
grafik

**Comment by amanape** [comment]
@tobitege Thanks a lot for taking the time to provide a thorough review! This is all great feedback that I'll be adding to the backlog and pick them off one-by-one. To answer your questions:

> the resizing grips will be re-added later?

Yes

> more of a "long time" quirk: the "send message" icon looks like an "upload" image to me, is there something closer to resembling an ENTER key?

You make a good point. I'll discuss this with Paul

> the assistant's chat replies do not get any border (anymore)?

No. They are new designs but we're still exploring alternatives here (or maybe go back to what it was).

> Start screen says "Attach a file" -> "Attach an image"?

Woops!

> Isn't "awaiting user input" normally in blue?

Thanks for pointing this out

For the most part, these issues are non blocking and can be addressed in follow-up PRs (this one has already grown larger than it should have). Just a heads up that this may merge without all of these points resolved yet.

**Comment by rbren** [comment]
Thanks for the feedback @tobitege!

I agree with Stephan that some of these can be pushed into follow-up issues, especially the "long running" stuff :)

The only one that sounds like a blocker to me is the settings modal bug--that seems worrisome

**Comment by tobitege** [comment]
Yes, it's mostly cosmetics I observed, no blockers to me. 😃

**Review by tobitege — APPROVED** [review]
LGTM! Quite a huge effort in this, thanks so much for all the work! 🥇
